File: UKParliament.CodeTest.Web/ClientApp/src/app/models/person-view-model.ts
export class PersonViewModel {
  id?: number | null;
  firstName: string;
  lastName: string;
  dob: Date | string; // Change the type to Date | string
  department: string;

  constructor(id: number | null, firstName: string, lastName: string, dob: Date | string, department: string) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.dob = dob;
    this.department = department;
  }
}

File: UKParliament.CodeTest.Web/ViewModels/PersonViewModel.cs
﻿namespace UKParliament.CodeTest.Web.ViewModels;

public class PersonViewModel
{
    public int Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public DateTime DOB { get; set; }
    public int DepartmentId { get; set; }
    public string DepartmentName { get; set; } = string.Empty;
}